src/functions.php

```diff
@@ -293,3 +291,26 @@ function _checkTypehint(callable $callback, \Throwable $reason): bool
     // If we look for a full match (intersection) and did not return early, we matched all types and are true
     return $isTypeUnion ? false : true;
 }
+
+/**
+ * @internal
+ */
+function matchesType($reason, $type): bool
```

Comment:
I see why this was done, but not a big fan of `@internal` functions here as they still clutter the namespace and may interfere with autocompletion. I consider this more a matter of preference, but do you think it's reasonable to inline the function definition considering it's not referenced anywhere else?

Reply:
Hey, I extracted these two functions because `matchesAllTypes` calls `matchesType` and wanted to avoid code duplication. But if you're ok with a bit of duplication to prevent polluting the global namespace, I'll inline them.

Reply:
Agree it's a tradeoff, so I guess use your own best judgement? Personally, I tend to accept some duplication as long as it's to a reasonable degree and does not repeat any business decisions (DRY).

Reply:
Done